src/libs/svg/lv_svg_render.c

```diff
@@ -1730,6 +1730,154 @@ static void _get_tspan_bounds(const lv_svg_render_obj_t * obj, lv_area_t * area)
+    }
+}
+
+static void _get_viewport_size(const struct _lv_svg_render_obj * obj, uint32_t * size)
+{
+    _get_obj_size(obj, size);
+    *size += sizeof(float) * 2;
+    *size += sizeof(bool);
+}
+
+static void _get_rect_size(const struct _lv_svg_render_obj * obj, uint32_t * size)
+{
+    _get_obj_size(obj, size);
+    *size += sizeof(float) * 6;
+}
+
+static void _get_circle_size(const struct _lv_svg_render_obj * obj, uint32_t * size)
+{
+    _get_obj_size(obj, size);
+    *size += sizeof(float) * 3;
+}
+
+static void _get_ellipse_size(const struct _lv_svg_render_obj * obj, uint32_t * size)
+{
+    _get_obj_size(obj, size);
+    *size += sizeof(float) * 4;
+}
+
+static void _get_line_size(const struct _lv_svg_render_obj * obj, uint32_t * size)
+{
+    _get_obj_size(obj, size);
+    *size += sizeof(float) * 4;
+}
+
+static void _get_poly_size(const struct _lv_svg_render_obj * obj, uint32_t * size)
+{
+    _get_obj_size(obj, size);
+    lv_svg_render_poly_t * poly = (lv_svg_render_poly_t *)obj;
+    lv_vector_path_t * path = poly->path;
+
+    *size += _calc_path_data_size(path);
+    *size += sizeof(lv_vector_path_t);
+    *size += sizeof(lv_area_t);
+    *size += sizeof(void *);
+}
+
+static void _get_use_size(const struct _lv_svg_render_obj * obj, uint32_t * size)
+{
+    _get_obj_size(obj, size);
+    lv_svg_render_use_t * use = (lv_svg_render_use_t *)obj;
+    if(use->xlink) {
+        *size += lv_strlen(use->xlink);
+    }
+    *size += sizeof(float) * 2;
+    *size += sizeof(void *);
+
+}
+
+static void _get_image_size(const struct _lv_svg_render_obj * obj, uint32_t * size)
+{
+    _get_obj_size(obj, size);
+    *size += sizeof(lv_draw_image_dsc_t);
+    *size += sizeof(lv_svg_aspect_ratio_t);
+    *size += sizeof(float) * 4;
+}
+
+static void _get_solid_size(const struct _lv_svg_render_obj * obj, uint32_t * size)
+{
+    _get_obj_size(obj, size);
+    *size += sizeof(lv_color_t);
+    *size += sizeof(float);
+}
+
+static void _get_grad_size(const struct _lv_svg_render_obj * obj, uint32_t * size)
+{
+    _get_obj_size(obj, size);
+    *size += sizeof(lv_vector_gradient_t);
+    *size += sizeof(lv_svg_gradient_units_t);
+}
+
+static void _get_span_size(const struct _lv_svg_render_obj * obj, uint32_t * size)
+{
+    _get_obj_size(obj, size);
+    lv_svg_render_tspan_t * span = (lv_svg_render_tspan_t *)obj;
+    if(span->family) {
+        *size += lv_strlen(span->family);
+    }
+
+    *size += _calc_path_data_size(span->path);
+    *size += sizeof(float);
+    *size += sizeof(lv_freetype_font_style_t);
+    *size += sizeof(lv_vector_path_t);
+    *size += sizeof(lv_area_t);
+    *size += sizeof(void *) * 3;
+}
+
+static void _get_txt_size(const struct _lv_svg_render_obj * obj, uint32_t * size)
+{
+    _get_obj_size(obj, size);
+    lv_svg_render_text_t * txt = (lv_svg_render_text_t *)obj;
+    if(txt->family) {
+        *size += lv_strlen(txt->family);
+    }
+
+    *size += lv_array_capacity(&txt->contents) * sizeof(void *);
```

Comment:
I think you meant to use `lv_array_size` everywhere you used `lv_array_capacity` in this file.

Reply:
No, I actually want to calculate the actual memory size, not the number of elements. For example, an array containing 3 pointer variables may actually have 4 solts.

Reply:
Okay. I misunderstood what size was being measured. It's a heuristic for the sum of allocated bytes due to the SVG? Out of curiosity, what's a use case of checking `draw_buf->data_size`? For debugging and info?